src/platform_posix.cc

```diff
@@ -139,8 +139,8 @@ std::string GetExecutablePath() {
   return std::string(path);
 #else
   char buffer[PATH_MAX] = {0};
-  readlink("/proc/self/exe", buffer, PATH_MAX);
-  return std::string(buffer);
+  assert(-1 != readlink("/proc/self/exe", buffer, PATH_MAX));
```

Comment:
Move readlink out of assert, since it will not be executed if assert is disabled.

Reply:
Good catch. Did you want to avoid the vulnerability when the pathname is longer than PATH_MAX? (readlink(2): "readlink() does not append a null byte to buf.  It  will .......")